pkg/util/deployment/deployment.go

```diff
@@ -217,3 +219,57 @@ func Revision(rc *api.ReplicationController) (int64, error) {
 	}
 	return strconv.ParseInt(v, 10, 64)
 }
+
+func IsRollingUpdate(deployment *extensions.Deployment) bool {
+	return deployment.Spec.Strategy.Type == extensions.RollingUpdateDeploymentStrategyType
+}
+
+// maxSurgeReplicas returns the maximum number of pods that can be scheduled above the original number of
+// pods from calculating deployment's maxSurge. Note that only deployments with RollingUpdate strategy
+// contains maxSurge.
+func maxSurgeReplicas(deployment *extensions.Deployment) (int, error) {
+	if !IsRollingUpdate(deployment) {
+		return 0, fmt.Errorf("Expect strategy type: %v, got %v", extensions.RollingUpdateDeploymentStrategyType, deployment.Spec.Strategy.Type)
+	}
+	maxSurge, isPercent, err := util.GetIntOrPercentValue(&deployment.Spec.Strategy.RollingUpdate.MaxSurge)
```

Comment:
Please factor this into a function. We do this in lots of places. util.GetValueFromIntOrPercent(string, int) or something similar.